core-bundle/contao/drivers/DC_Table.php

```diff
@@ -3974,12 +3975,13 @@ public function ajaxTreeView($id, $level)
 			$arrIds[] = $objRows->id;
 		}
 
-		$arrClipboard = System::getContainer()->get('contao.data_container.clipboard_manager')->get($this->strTable);
+		$clipboardManager = System::getContainer()->get('contao.data_container.clipboard_manager');
+		$arrClipboard = $clipboardManager->get($this->strTable);
 		$blnClipboard = null !== $arrClipboard;
 
 		for ($i=0, $c=\count($arrIds); $i<$c; $i++)
 		{
-			$return .= ' ' . trim($this->generateTree($table, $arrIds[$i], array('p'=>($arrIds[$i - 1] ?? null), 'n'=>($arrIds[$i + 1] ?? null)), $hasSorting, $margin, $blnClipboard ? $arrClipboard : false, $arrClipboard !== null && ($id == $arrClipboard['id'] || (\is_array($arrClipboard['id']) && \in_array($id, $arrClipboard['id'])) || (!$blnPtable && !\is_array($arrClipboard['id']) && \in_array($id, $db->getChildRecords($arrClipboard['id'], $table)))), $blnProtected));
+			$return .= ' ' . trim($this->generateTree($table, $arrIds[$i], array('p'=>($arrIds[$i - 1] ?? null), 'n'=>($arrIds[$i + 1] ?? null)), $hasSorting, $margin, $blnClipboard ? $arrClipboard : false, $clipboardManager->isCircularReference($this->strTable, $arrIds[$i], !$blnPtable), $blnProtected));
```

Comment:
Here the original line uses `$table` but you use `$this->strTable` now (also see my comment above).

Reply:
Changed again in 22003365076c5ae768bc1f014697107d0415c591 😂 

- for `tl_article` (extended tree mode), `$table` contains `tl_page` and `$this->strTable` is `tl_article`. However, the clipboard always refers to the current table (`tl_article`), and checking for recursion on `tl_page` with the clipboard having an ID of `tl_article` makes little sense to me.

Reply:
No, sorry, this cannot work. `$arrIds` also contains the IDs of `$table` and not of `$this->strTable`, so now you provide two arguments that point to different tables.

Reply:
than's true in theory, but stupid in practice 😂 
- when rendering `tl_page` (a tree view), it does not matter since `$table === $this->strTable`
- when rendering `tl_article` (an extended tree), there can never be a circular reference because the clipboard IDs are not nested. Passing `$this->strTable` will cause the method to fail in checking for `MODE_TREE` and return false before actually checking the given ID

Reply:
> and return false before actually checking the given ID

Where exactly should this be? The method returns `string` and there is only one early return right at the beginning of the method.

Reply:
As discussed in the call, the changes are correct 🎉